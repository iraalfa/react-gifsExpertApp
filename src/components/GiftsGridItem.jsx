import React from 'react'
import PropTypes from 'prop-types';
import 'animate.css'

const GiftsGridItem = ({title, url}) => {
    // console.log(img)
    return (
        <div className="card animate__animated.animate__flash">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GiftsGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

export default GiftsGridItem
