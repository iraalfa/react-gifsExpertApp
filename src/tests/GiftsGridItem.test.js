import { shallow } from "enzyme"
import GiftsGridItem from "../components/GiftsGridItem"

describe('Pruebas en <GiftsGridItem />', () => {
    const title = 'A title'
    const url = 'http://localhost/something.jpg'
    const wrapper = shallow(<GiftsGridItem  title={title} url={url}/>)

    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p')

        expect(p.text().trim()).toBe(title)
    })

    test('Debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img')
        // console.log(img.props())
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('debe tener animate.animate_flash', () => {

        const div = wrapper.find('div')
        const className = div.prop('className')

        expect(className.includes('animate.animate_flash')).toBe(false)
        console.log(div.props('className'));
    });

})