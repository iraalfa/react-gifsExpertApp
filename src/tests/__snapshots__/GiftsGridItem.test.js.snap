// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftsGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated.animate__flash"
>
  <img
    alt="A title"
    src="http://localhost/something.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
